### PR TITLE
Update 2023 and 2024 Heavy Ion relval workflows for 2024 HLT menu

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

PR https://github.com/cms-sw/cmssw/pull/46375 updated the `HIon` HLT menu such that it requires a L1T menu ([`L1Menu_CollisionsHeavyIons2024_v1_0_5_xml`](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/tags/L1Menu_CollisionsHeavyIons2024_v1_0_5_xml) ) different from the one used in 2023 (more L1T seeds are required by trigger paths introduced in the meanwhile, see [link to payload inspector](https://cern.ch/526sr) [1]).
The goal of this PR is to update the 2023 HI relval workflow to use the `@fake2` menu and to update the 2024 HI workflows to use the updated `HIon` menu (consequently the DQM and validation steps are updated to run the HLT sequences).

#### PR validation:

Tested successfully 
```
runTheMatrix.py -l 160,160.02,160.03,160.1,160.2,160.3,160.4,160.501,161,161.02,161.03,161.1,161.2,161.3,161.4,180.0,180.1,181,181.1 --ibeos
```

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, will be backported to `CMSSW_14_1_X`

---
[1]

<details>
  <summary>L1T menu comparison</summary>

image

</details>

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflows = 160, 160.02, 160.03, 160.1, 160.2, 160.3, 160.4, 161, 161.02, 161.03, 161.1, 161.2, 161.3, 161.4, 160.501, 180.0, 180.1, 181, 181.1

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
urgent
   * fixes IB relval failures

**Comment by mandrenguyen** [comment]
@cms-sw/pdmv-l2 @cms-sw/upgrade-l2 please have a look at this and the 14_1_X backport.

**Comment by Moanwar** [comment]
+Upgrade

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by mandrenguyen** [comment]
+1

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -2045,10 +2045,10 @@ def lhegensim2018ml(fragment,howMuch):
 #steps['RESIMDIGI']=merge([{'-s':'reGEN,reSIM,DIGI,L1,DIGI2RAW,HLT:@fake,RAW2DIGI,L1Reco','-n':10,'--restoreRNDSeeds':'','--process':'HLT'},steps['DIGI']])
 
 
-steps['DIGIHI2024APPROXCLUSTERS']=merge([{'-s':'DIGI:pdigi_hi_nogen,L1,DIGI2RAW,HLT:@fake2'}, hiDefaults2024_approxClusters, {'--pileup':'HiMixNoPU'}, step2Upg2015Defaults])
-steps['DIGIHI2024']=merge([{'-s':'DIGI:pdigi_hi_nogen,L1,DIGI2RAW,HLT:@fake2'}, hiDefaults2024, {'--pileup':'HiMixNoPU'}, step2Upg2015Defaults])
+steps['DIGIHI2024APPROXCLUSTERS']=merge([{'-s':'DIGI:pdigi_hi_nogen,L1,DIGI2RAW,HLT:HIon'}, hiDefaults2024_approxClusters, {'--pileup':'HiMixNoPU'}, step2Upg2015Defaults])
+steps['DIGIHI2024']=merge([{'-s':'DIGI:pdigi_hi_nogen,L1,DIGI2RAW,HLT:HIon'}, hiDefaults2024, {'--pileup':'HiMixNoPU'}, step2Upg2015Defaults])
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/relval_steps.py:2049]: Looks like `DIGIHI2024MIX` should have been updated too [log](https://cmssdt.cern.ch/SDT/html/cmssdt-ib/#/relVal/CMSSW_14_2/2024-10-16-2300?selectedArchs=el8_amd64_gcc12&selectedFlavors=ROOT6_X&selectedStatus=failed).

Comment (mmusich) [reply Configuration/PyReleaseValidation/python/relval_steps.py:2049]: Followed up in https://github.com/cms-sw/cmssw/pull/46418

**Pull request merged**